Userland/Libraries/LibGUI/FilePicker.cpp

```diff
@@ -301,4 +329,34 @@ void FilePicker::set_path(String const& path)
     m_model->set_root_path(new_path);
 }
 
+String FilePicker::history_file_path() const
+{
+    return LexicalPath::canonicalized_path(String::formatted("{}/FilePicker-{}.txt", Core::StandardPaths::history_directory(), m_mode == Mode::OpenFolder ? "OpenFolder" : "Open"));
+}
+
+ErrorOr<void> FilePicker::load_history()
+{
+    m_history.clear();
+    auto file = TRY(Core::File::open(history_file_path(), Core::OpenMode::ReadOnly));
+
+    while (!file->eof()) {
+        auto line = file->read_line();
+        if (!line.is_empty() && !line.is_whitespace())
+            m_history.append(line);
+    }
+
+    return {};
+}
+
+ErrorOr<void> FilePicker::save_history()
+{
+    auto directory = TRY(Core::Directory::create(Core::StandardPaths::history_directory(), Core::Directory::CreateDirectories::Yes));
+    auto file = TRY(Core::File::open(history_file_path(), Core::OpenMode::WriteOnly));
```

Comment:
Why don't you use the `Core::Directory` API for directly opening a file in the directory?